lib/common-yargs.js

```diff
@@ -1,106 +1,200 @@
+const fs = require('fs')
+const parseGitHubURL = require('./parse-github-url')
+
 /**
  * These descriptions are exported so that they can be used to group global options in usage output.
  */
 const descriptions = {
-	token: "GitHub personal access token (uses `GITHUB_PERSONAL_ACCESS_TOKEN` environment variable by default). Generate one at https://github.com/settings/tokens.",
-	json: "Format command output as JSON string",
+	token: 'GitHub personal access token. Generate one at https://github.com/settings/tokens.',
+	json: 'Format command output as JSON string',
 }
 
-const withToken = yargs => yargs.option("token", {
-
-	// IMPORTANT: We use a function here so the token is not output on the command line.
-	default: () => process.env.GITHUB_PERSONAL_ACCESS_TOKEN,
-	describe: descriptions.token,
-	type: "string",
-	coerce: value => {
-		if (!value) {
-			throw new Error(
-				"Error: `--token` option is required as a valid GITHUB_PERSONAL_ACCESS_TOKEN environment variable does not exist"
+const withToken = options => yargs =>
+	yargs
+		.option(
+			'token',
+			Object.assign(
+				{},
+				{
+					alias: 't',
+					describe: descriptions.token,
+					type: 'string',
+				},
+				options
 			)
-		}
-		return value
-	}
-})
-
-const withJson = yargs => yargs.option("json", {
-	describe: descriptions.json,
-	type: "boolean",
-})
+		)
+		.check(argv => {
+			if (!argv.token && !process.env.GITHUB_PERSONAL_ACCESS_TOKEN) {
+				throw new Error('Missing required argument: token.')
+			}
+			return true
+		})
+		.default('token', process.env.GITHUB_PERSONAL_ACCESS_TOKEN, 'process.env.GITHUB_PERSONAL_ACCESS_TOKEN')
```

Comment:
Note: The GITHUB_PERSONAL_ACCESS_TOKEN is still not output to the commandline. ✅